src/tyr/route_serializer_osrm.cc

```diff
@@ -549,6 +549,14 @@ json::ArrayPtr intersections(const valhalla::DirectionsLeg::Maneuver& maneuver,
       edges.emplace_back(((prior_heading + 180) % 360), entry, true, false);
     }
 
+    if (i > 0 && !arrive_maneuver) {
```

Comment:
I would at it to the code where "main" outgoing edge is added
```cpp
// Add the edge departing the node
    if (!arrive_maneuver) {
      edges.emplace_back(curr_edge->begin_heading(), true, false, true);
      // add other outgoing edges
      ...
    }
```

Reply:
the logic here is different, we are adding intersecting edges that are not starting or not ending, only in the middle of the road.